### PR TITLE
[terraform-provider-docs] - clean up and improve local provider usage…

### PR DESCRIPTION
**Pull request by ScottSuarez**
… documentation improvement 
Supplied more ways to utilize local provider and cleaned up documentation.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -73,22 +73,27 @@ resource.TestStep{
...
-ln -s $GOPATH/bin/terraform-provider-google ~/.terraform.d/plugins/terraform-provider-google
-ln -s $GOPATH/bin/terraform-provider-google-beta ~/.terraform.d/plugins/terraform-provider-google-beta
-```
+### Using a release version of terraform
+2. When you run `terraform init` supply the binary output location using the [-plugin-dir](https://www.terraform.io/docs/commands/init.html#plugin-dir-path) parameter. ie `terraform init -plugin-dir=$GOPATH/bin`
+    * note: terraform will not download additional providers remotely. All requested providers should be in the -plugin-dir
+    * note: each time you rebuild the binary you will have to rerun `terraform init -plugin-dir=<your-binary-location>` as the hash is invalidated.
+3. Use the [provider discovery directory](https://www.terraform.io/docs/extend/how-terraform-works.html#discovery) at `~/.terraform.d/plugins`. Terraform will attempt to use the binaries here as a provider before trying to pull the provider remotely.
+    * note: you can either copy the provider binary to this location or do a symlink to the build output
+    * note: you will need to delete this binary or syslink to pull the provider remotely
```
Comment (rileykarson) [inline .github/CONTRIBUTING.md:87]:
```suggestion
    * note: you will need to delete this binary or symlink to pull the provider remotely implicitly (if you specify a `version` constraint, a remote provider will always be used)
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -73,22 +73,27 @@ resource.TestStep{
...
+      ```bash
+      # helpful bash to system link to binary path.
+      # this allows the binary to stay in sync whenever you run make build
+      ln -s $GOPATH/bin/terraform-provider-google ~/.terraform.d/plugins/terraform-provider-google
+      ln -s $GOPATH/bin/terraform-provider-google-beta ~/.terraform.d/plugins/terraform-provider-google-beta
+      ```
+### FAQ
+* Why isn't it using my local provider?? I did everything right!
+  * If you've already used a release version of a provider in a given directory by running `terraform init`, Terraform will not use the locally built copy; remove the release version from the `./.terraform/` to start using your locally build copy.
```
Comment (rileykarson) [inline .github/CONTRIBUTING.md:96]:
```suggestion
  * If you've already used a release version of a provider in a given directory by running `terraform init`, Terraform will not use the locally built copy; remove the release version from the `./.terraform/` to start using your locally built copy.
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -73,22 +73,27 @@ resource.TestStep{
 },
 ```
 
-## Running a local copy of the provider
-If you are building Terraform from source with a Google provider built from source, Terraform will automatically use the
+## Instructing terraform to use a local copy of the provider
+There are three different consumption strategies you can utilize
```
Comment (rileykarson) [inline .github/CONTRIBUTING.md:77]: nit: should we add some spaces between headers and text here (and below)?

**Review by rileykarson — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (ScottSuarez) [reply .github/CONTRIBUTING.md:77]: Changed header to bullet list. looks slightly better but not sure what I like more. I tried adding the spacing below/above this text. Looked kind of funky

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (ScottSuarez) [reply .github/CONTRIBUTING.md]: fixed

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (ScottSuarez) [reply .github/CONTRIBUTING.md:77]: unless you mean adding spacing purely for the reader of the raw readme files

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (ScottSuarez) [reply .github/CONTRIBUTING.md]: done

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (ScottSuarez) [reply .github/CONTRIBUTING.md:77]: resolving for now. feel free to reopen if you have strong opinion.

**Comment by ScottSuarez** [comment]
Waiting till EoD to complete to give the opportunity for more feedback

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (rileykarson) [reply .github/CONTRIBUTING.md:77]: Oh, I just meant the raw markdown!

**Pull request merged**